src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -183,66 +174,88 @@ function addScriptTagsToDocument(resourceLoader: WebAssemblyResourceLoader) {
     .keys(resourceLoader.bootConfig.resources.runtime)
     .filter(n => n.startsWith('dotnet.') && n.endsWith('.js'))[0];
   const dotnetJsContentHash = resourceLoader.bootConfig.resources.runtime[dotnetJsResourceName];
-  const scriptElem = document.createElement('script');
-  scriptElem.src = `_framework/${dotnetJsResourceName}`;
-  scriptElem.defer = true;
-
-  // For consistency with WebAssemblyResourceLoader, we only enforce SRI if caching is allowed
-  if (resourceLoader.bootConfig.cacheBootResources) {
-    scriptElem.integrity = dotnetJsContentHash;
-    scriptElem.crossOrigin = 'anonymous';
-  }
+  let src = `_framework/${dotnetJsResourceName}`;
 
   // Allow overriding the URI from which the dotnet.*.js file is loaded
   if (resourceLoader.startOptions.loadBootResource) {
     const resourceType: WebAssemblyBootResourceType = 'dotnetjs';
-    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, scriptElem.src, dotnetJsContentHash);
+    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, src, dotnetJsContentHash);
     if (typeof (customSrc) === 'string') {
-      scriptElem.src = customSrc;
+      src = customSrc;
     } else if (customSrc) {
-      // Since we must load this via a <script> tag, it's only valid to supply a URI (and not a Request, say)
+      // Since we must load this via a import, it's only valid to supply a URI (and not a Request, say)
       throw new Error(`For a ${resourceType} resource, custom loaders must supply a URI string.`);
     }
   }
 
-  document.body.appendChild(scriptElem);
-}
-
-// Due to a strange behavior in macOS Catalina, we have to delay loading the WebAssembly files
-// until after it finishes evaluating a <script> element that assigns a value to window.Module.
-// This may be fixed in a later version of macOS/iOS, or even if not it may be possible to reduce
-// this to a smaller workaround.
-function addGlobalModuleScriptTagsToDocument(callback: () => void) {
-  const scriptElem = document.createElement('script');
+  // For consistency with WebAssemblyResourceLoader, we only enforce SRI if caching is allowed
+  if (resourceLoader.bootConfig.cacheBootResources) {
+    try {
+      if (globalThis.navigator) {
+        const nav: any = globalThis.navigator;
+        if (nav.userAgentData && nav.userAgentData.brands && nav.userAgentData.brands.some((i: any) => i.brand === 'Chromium')) {
+          const scriptElem = document.createElement('link');
+          scriptElem.rel = 'modulepreload';
+          scriptElem.href = src;
+          // it will make dynamic import fail if the hash doesn't match
+          scriptElem.integrity = dotnetJsContentHash;
+          scriptElem.crossOrigin = 'anonymous';
+          document.head.appendChild(scriptElem);
+        }
+      }
+    } catch (err) {
+      console.error(err);
```

Comment:
I expect this was left over from development. We should remove the `try`/`catch` and the logging to console, because if there's an unhandled error we'd prefer the promise chain to be rejected and have the natural unhandled error behavior.

Reply:
will do